sssd_test_framework/utils/adcli.py

```diff
@@ -103,6 +103,41 @@ def testjoin(
 
         return self.host.conn.exec(["adcli", "testjoin", domain, *args], raise_on_error=False)
 
+    def update(
+        self,
+        *,
+        domain: str,
+        args: list[str] | None = None,
+        password: str,
+        login_user: str,
+    ) -> ProcessResult:
+        """
+        Update a computer account's password, and other attributes.
+
+        :param domain: Domain.
+        :type domain: str
+        :param args: Additional arguments, defaults to None
+        :type args: list[str] | None, optional
+        :param password: Password
+        :type password: str
+        :param login_user: Authenticating User
+        :type login_user: str
+        :return: Result of called command.
+        :rtype: ProcessResult
+        """
+        if args is None:
+            args = []
+
+        return self._exec_adcli(
+            subcommand="update",
+            positional_args=[],
+            domain=domain,
+            password=password,
+            login_user=login_user,
+            krb=True,
```

Comment:
Hi,

if I understand it correctly with `krb=True` `_exec_adcli()` will unconditionally call `kinit` with the given `login_user` and `password`. But the typical use case for `adcli update` is to be called without user name and password because it will use a keytab. So I would expect this method to fail if no user name and password is provided. Maybe it would be easier to call `adcli` directly here, similar to `info` or `testjoin` methods.

bye,
Sumit